### PR TITLE
Clarify stitching log message (#2507)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain            <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox`                             <sub>or added `no sandbox` label</sub>
- [x] Reindexed `sandbox`                                   <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit
- [x] Moved linked issue to Merged
- [ ] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain                                       <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations                        <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Primary reviewer (reindex) 

- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Deleted BQ slot committment in `dev`                  <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Deleted BQ slot committment in `prod`                 <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing `prod`</sub>

Primary reviewer

- [ ] Unassign PR from reviewer and assign to PM

### CODE REVIEW

**Review by danielsotirhos — APPROVED** [review]
Looks clean.

File: src/azul/plugins/repository/tdr/__init__.py [hunk 1]
```diff
@@ -241,10 +241,12 @@ def _stitch_bundles(self,
...
             else:
                 log.debug('Bundle %r is self-contained', bundle)
-        if log.isEnabledFor(logging.DEBUG):
-            log.debug('Stitched together bundles: %r', processed)
-        else:
-            log.info('Stitched together %i bundles', len(processed))
+        processed.remove(root_bundle)
+        if processed:
+            if log.isEnabledFor(logging.DEBUG):
+                log.debug('Stitched bundles: %r', processed)
```
Comment (hannes-ucsc) [inline src/azul/plugins/repository/tdr/__init__.py:247]:
```suggestion
                log.debug('Stitched %i bundles: %r', len(processed), processed)
```

and think about ways to reduce the resulting code duplication.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: src/azul/plugins/repository/tdr/__init__.py [hunk 2]
```diff
@@ -241,10 +241,12 @@ def _stitch_bundles(self,
                 unprocessed |= self._find_upstream_bundles(dangling_inputs) - processed
             else:
                 log.debug('Bundle %r is self-contained', bundle)
-        if log.isEnabledFor(logging.DEBUG):
-            log.debug('Stitched together bundles: %r', processed)
-        else:
-            log.info('Stitched together %i bundles', len(processed))
+        processed.remove(root_bundle)
+        if processed:
+            msg = f'Stitched {len(processed)} bundles'
```
Comment (hannes-ucsc) [inline src/azul/plugins/repository/tdr/__init__.py:246]:
```suggestion
            msg = f'Stitched {len(processed)} bundle(s)'
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
How about this? I am undecided.

```diff
Index: src/azul/plugins/repository/tdr/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/plugins/repository/tdr/__init__.py b/src/azul/plugins/repository/tdr/__init__.py
--- a/src/azul/plugins/repository/tdr/__init__.py	(revision 48d36bd16bd8023b51ee21e0061e2ff91f56935a)
+++ b/src/azul/plugins/repository/tdr/__init__.py	(date 1611194036092)
@@ -243,10 +243,8 @@
                 log.debug('Bundle %r is self-contained', bundle)
         processed.remove(root_bundle)
         if processed:
-            msg = f'Stitched {len(processed)} bundles'
-            if log.isEnabledFor(logging.DEBUG):
-                msg += f': {processed!r}'
-            log.info(msg)
+            args = (': ', processed) if log.isEnabledFor(logging.DEBUG) else ('', '')
+            log.info('Stitched together %i bundle(s)%s%s', len(processed), *args)
         return entities, links_jsons
 
     def _retrieve_links(self, links_id: BundleFQID) -> JSON:
```

File: src/azul/plugins/repository/tdr/__init__.py (hunk 2 above)
Comment (noah-aviel-dove) [reply src/azul/plugins/repository/tdr/__init__.py:246]: I think that's an improvement. What do you think about 
```
arg = f': {processed!r}' if log.isEnabledFor(logging.DEBUG) else ''
log.info('Stitched %i bundle(s)%s', len(processed), arg)
``` 
to preserve the `%r` behavior?

Comment (hannes-ucsc) [reply src/azul/plugins/repository/tdr/__init__.py:246]: Yes, let's go with that.

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
For demo, show a sample log message from `dev`. Might have to wait until `dev` has been reindexed.

@DataBiosphere/azul-operators, I set the `reindex` label to indicate that this PR *should* be batched with another one that strictly requires reindexing.

**Pull request merged**